packages/babylon/src/plugins/typescript.js

```diff
@@ -1866,6 +1866,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           );
         case "TSParameterProperty":
           return super.toAssignable(node, isBinding, contextDescription);
+        case "TSAsExpression":
+          return super.toAssignable(
```

Comment:
I think that here it should be `thsi.toAssignable`, to work with nested as expressions. e.g. `((a as T) as U) = b;`

Reply:
Yes! I missed that case. I'll make the change.